core/xchainmg_net.go

```diff
@@ -90,12 +90,9 @@ func (xm *XChainMG) handleReceivedMsg(msg *xuper_p2p.XuperMessage) {
 }
 
 func (xm *XChainMG) handlePostTx(msg *xuper_p2p.XuperMessage) {
-	txStatus := &pb.TxStatus{}
-
-	// Unmarshal msg
-	err := proto.Unmarshal(msg.GetData().GetMsgInfo(), txStatus)
-	if err != nil {
-		xm.Log.Error("handlePostTx Unmarshal msg to tx error", "logid", msg.GetHeader().GetLogid())
+	txStatus, err := handlePostTxMsgFromRemoteNode(msg)
```

Comment:
It's better to wrap compress process in p2p library, you can leverage p2pv2.MessageOption to provide WithCompress option for SendMessage/SendMessageWithResponse.